subsys/bluetooth/controller/hci/hci.c

```diff
@@ -2602,7 +2603,10 @@ static void le_df_set_cl_iq_sampling_enable(struct net_buf *buf, struct net_buf
 						 cmd->switch_pattern_len,
 						 cmd->ant_ids);
 
-	*evt = cmd_complete_status(status);
+	rp = hci_cmd_complete(evt, sizeof(*rp));
+
+	rp->status = status;
+	rp->sync_handle = sync_handle;
```

Comment:
`cpu_to_le16(sync_handle)`?

Reply:
Ah, thanks! I'll change it to `rp->sync_handle = cmd->sync_handle;`

Reply:
I'm not sure how the command-response buffer handling is today, you might want to be a bit careful with that if the command buffer gets reused for the event. In that case you might risk corruption if e.g. rp->status occupies the same memory as cmd->sync_handle. @carlescufi or @cvinayak should know that's safe to do in the controller.

Reply:
Corrected

Reply:
In a combined build the command complete event would be the same buffer as the command. `bt_buf_get_cmd_complete` in `subsys/bluetooth/host/buf.c`.

Reply:
I've figured it out when I run test code. Changed according to @joerchan suggestion https://github.com/zephyrproject-rtos/zephyr/pull/35788#discussion_r641742811